### PR TITLE
Support run/debug and unit test for VSCode

### PR DESCRIPTION
**Pull request by AlexeiShmidman**
# Additional Info
build Eclipse project structure to point to actual folders with source code and build files for:
- support Debugger for Java for VSCode
- support Java Test Runner for VSCode

### CODE REVIEW

File: com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/classpath/BazelClasspathContainer.java [hunk 1]
```diff
@@ -117,6 +123,71 @@ public static void clean() {
...
+        File javaToolsDir = new File(testRunnerDir, javaToolsPath);
+        if (!javaToolsDir.exists()) {
+            return null;
+        }
+        File runnerJar = new File(javaToolsDir, runnerJarName);
+        if (!runnerJar.exists()) {
+            return null;
+        }
+        return BazelJdtPlugin.getJavaCoreHelper()
+                .newLibraryEntry(org.eclipse.core.runtime.Path.fromOSString(runnerJar.toString()), null, null);
```
Comment (kkazakevich) [inline com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/classpath/BazelClasspathContainer.java:150]: maybe import static method?

File: com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java [hunk 2]
```diff
@@ -466,6 +509,19 @@ public void runBazelVersionCheck() throws BazelCommandLineToolConfigurationExcep
...
+        if (SystemUtils.IS_OS_MAC) {
+            return "darwin";
+        }
+        if (SystemUtils.IS_OS_WINDOWS) {
+            return "windows";
+        }
+        if (SystemUtils.IS_OS_LINUX) {
+            return "linux";
+        }
+        return null;
```
Comment (kkazakevich) [inline com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java:522]: maybe throw exception or return empty string to avoid NPE in case of unexpected OS?

File: com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java [hunk 3]
```diff
@@ -261,6 +266,45 @@ public File getBazelWorkspaceBin(WorkProgressMonitor progressMonitor) {
...
 
+    public File getBazelGeneratedFilesFolder() {
+        try {
+            ImmutableList.Builder<String> argBuilder = ImmutableList.builder();
+            argBuilder.add("info").add("bazel-genfiles");
+
+            List<String> outputLines = bazelCommandExecutor.runBazelAndGetOutputLines(bazelWorkspaceRootDirectory, null,
+                argBuilder.build(), (t) -> t);
+            outputLines = BazelCommandExecutor.stripInfoLines(outputLines);
+            File dir = new File(String.join("", outputLines));
```
Comment (kkazakevich) [inline com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java:277]: seems no need to declare dir variable

File: com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java [hunk 4]
```diff
@@ -466,6 +508,19 @@ public void runBazelVersionCheck() throws BazelCommandLineToolConfigurationExcep
...
 
+    public String getOperatingSystemDirectoryName() {
+        if (SystemUtils.IS_OS_MAC) {
+            return "darwin";
+        }
+        if (SystemUtils.IS_OS_WINDOWS) {
+            return "windows";
+        }
+        if (SystemUtils.IS_OS_LINUX) {
+            return "linux";
```
Comment (PMitrafanau) [inline com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java:519]: Isn't it `k8` sometimes?

**Review by PMitrafanau — CHANGES_REQUESTED** [review]

File: com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/classpath/BazelClasspathContainer.java [hunk 5]
```diff
@@ -117,10 +129,74 @@ public static void clean() {
...
+        return cpEntry;
+    }
+
+    private List<IClasspathEntry> calculateBazelClasspath(Collection<AspectPackageInfo> aspectPackageInfos,
+            WorkProgressMonitor progressMonitor) {
+        BazelWorkspaceCommandRunner bazelWorkspaceCmdRunner = getBazelCommandManager()
+                .getWorkspaceCommandRunner(getBazelWorkspaceRootDirectory());
+        // filter-out main modules and
+        List<IClasspathEntry> classpathEntries = aspectPackageInfos.stream() //
+                .filter((AspectPackageInfo info) -> ObjectUtils.notEqual("java_library", info.getKind())
```
Comment (PMitrafanau) [inline com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/classpath/BazelClasspathContainer.java:178]: We could store these constants in a static collection (e.g. `Set`) and shorten this to:
`!COLLECTION_NAME.contains(info.getKind())`

File: com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java (hunk 4 above)
Comment (AlexeiShmidman) [reply com.salesforce.b2eclipse.jdt.ls/src/main/java/com/salesforce/b2eclipse/command/BazelWorkspaceCommandRunner.java:519]: it should be as a part of output path, not OS-dependent

**Review by PMitrafanau — APPROVED** [review]
